packages/wonder-blocks-toolbar/src/components/toolbar.tsx

```diff
@@ -81,54 +81,39 @@ export default function Toolbar({
         <View
             style={[
                 sharedStyles.container,
+                !title
+                    ? sharedStyles.containerWithNoTitle
+                    : typeof title === "string"
+                    ? sharedStyles.containerWithTextTitle
+                    : sharedStyles.containerWithNodeTitle,
                 color === "dark" && sharedStyles.dark,
                 size === "small" && sharedStyles.small,
             ]}
         >
-            <View
-                style={[
-                    sharedStyles.column,
-                    sharedStyles.leftColumn,
-                    title ? sharedStyles.withTitle : null,
-                ]}
-            >
-                {leftContent}
-            </View>
+            <View style={sharedStyles.leftColumn}>{leftContent}</View>
 
-            {title && typeof title === "string" ? (
-                <View style={[sharedStyles.column, sharedStyles.wideColumn]}>
-                    <View style={[sharedStyles.titles, sharedStyles.center]}>
-                        <TitleComponent id="wb-toolbar-title">
-                            {title}
-                        </TitleComponent>
-                        {subtitle && (
-                            <LabelSmall
-                                style={
-                                    color === "light" && sharedStyles.subtitle
-                                }
-                            >
-                                {subtitle}
-                            </LabelSmall>
-                        )}
-                    </View>
-                </View>
-            ) : (
-                // We don't use wideColumn here to allow more flexibility with
-                // the custom node.
-                <View style={[sharedStyles.column]}>
-                    <View style={[sharedStyles.titles]}>{title}</View>
+            {title && typeof title === "string" && (
+                <View style={sharedStyles.titles}>
+                    <TitleComponent id="wb-toolbar-title">
+                        {title}
+                    </TitleComponent>
+                    {subtitle && (
+                        <LabelSmall
+                            style={color === "light" && sharedStyles.subtitle}
+                        >
+                            {subtitle}
+                        </LabelSmall>
+                    )}
                 </View>
             )}
+            {title && typeof title !== "string" && (
+                <View style={sharedStyles.titles}>{title}</View>
+            )}
+            {!title && (
+                <View style={leftContent ? sharedStyles.spacer : undefined} />
+            )}
```

Comment:
**question:** I wonder if this could be solved by applying `gap: spacing.small_12` in the parent container. Or is this need to be flexible enough for responsiveness?